node/src/main.rs

```diff
@@ -574,6 +572,29 @@ fn async_main() -> impl Future<Item = (), Error = ()> + Send + 'static {
             .expect("Failed to start GraphQL subscription server"),
     );
 
+    // Periodically check for contention in the tokio threadpool. First spawn a
+    // task that simply responds to "ping" requests. Then spawn a separate
+    // thread to periodically ping it and check responsiveness.
+    let (ping_send, ping_receive) = mpsc::channel::<std::sync::mpsc::Sender<()>>(1);
+    tokio::spawn(
+        ping_receive
+            .for_each(move |pong_send| pong_send.clone().send(()).map(|_| ()).map_err(|_| ())),
+    );
+    let contention_logger = logger.clone();
+    std::thread::spawn(move || loop {
+        std::thread::sleep(Duration::from_secs(10));
+        let (pong_send, pong_receive) = std::sync::mpsc::channel();
+        ping_send.clone().send(pong_send).wait().unwrap();
+        let mut timeout = Duration::from_millis(1);
+        while pong_receive.recv_timeout(timeout).is_err() {
+            warn!(contention_logger, "Possible contention in tokio threadpool";
```

Comment:
If add a `LogCode` here, that would help in making a Grafana panel + alert for this

Reply:
I'll add a code, but we should probably alert on any `warn` level log.